packages/@react-aria/menu/src/useSubmenuTrigger.ts

```diff
@@ -0,0 +1,260 @@
+        if (!openTimeout.current) {
+          openTimeout.current = setTimeout(() => {
+            onSubmenuOpen();
+          }, 200);
+        }
+      } else if (!isHovered) {
+        cancelOpenTimeout();
+      }
+    }
+  };
+
+  let onBlur = (e) => {
+    // TODO: getting rid of the (!isElementInChildOfActiveScope(e.relatedTarget) part of the check below fixes hovering to open a root unavailable menu item when focus is on a submenu
+    // but breaks the case where the user is hovering over a submenu's submenu trigger child item and then hovers a root menu item (eg hover lvl 1 item 2 -> hover lvl 2 item 3 -> hover lvl 1 item 3)
+    // Ideally we'd be able to track the full menu tree and then check if focus has moved to an element that isn't part of the current submenu tree and close it then
+    if (state.isOpen && parentMenuRef.current.contains(e.relatedTarget)) {
+      onSubmenuClose();
+    }
+  };
+
+  // TODO: Track the external element being interacted with so that we can move focus to it when the subMenu closes. If we don't manually move focus, then focus gets lost to the body
+  // Calling ref.current.focus in useMenuItem's onHoverStart doesn't seem to move focus properly unfortunately
+  let closeTarget = useRef(null);
+  let shouldCloseOnInteractOutside = (target) => {
+    if (target !== ref.current) {
+      closeTarget.current = target;
+      return true;
+    }
+
+    return false;
+  };
+
+  // TODO: this onExit is Spectrum specific, so kinda weird to have in the aria hook. However, perhaps it would be nice if the aria Popover has onExit as well so
+  // users wouldn't need to reimplement this?
+  let onExit = () => {
+    // TODO: A bit awkward, but this first part of the if statement handles moving focus to the item the user hovers, specifically if said item is a menu item
+    // in one of the menu ancestors of the currently open submenu. The ideal logic would be to check if the closeTarget was in any of the menus in the tree and do nothing if so. Technically
+    // the else if part of this handles this already since hovering a submenu item sets it as the focusedKey and then focusing the submenu itself will move focus to the focusedKey via useSelectableCollection,
+    // but it breaks if the user hovers the base menu's first submenu trigger since then the first submenu won't actually close
+    // due to the logic in shouldCloseOnInteractOutside, and thus the first submenu retains focus, overriding the hover focus of the root menu's submenu trigger
+    if (closeTarget.current) {
+      closeTarget.current?.focus();
+      closeTarget.current = null;
+    } else if (!parentMenuRef.current.contains(document.activeElement)) {
+      // need to return focus to the trigger because hitting Esc causes focus to go to the subdialog, which is then unmounted
+      // this leads to blur never being fired nor focus on the body
+      parentMenuRef.current.focus();
+    }
+
+    cancelOpenTimeout();
+  };
```

Comment:
I'd like to understand better why this is needed. Shouldn't FocusScope handle restoring focus back to the trigger?

Reply:
So this is handling the following mouse hover case:
Case 1:
1. User has 3 levels of submenus open
2. User moves their mouse out of the last menu and in such a way that they are hovering a non-trigger menu item in the root menu.
3. We expect that all menus except the root menu would close and that focus would remain on the hovered menu item.

If we get rid of `onExit` and set `disableFocusManage=false`, then FocusScope will indeed handle restoring focus back to the trigger, breaking `Case 1` since focus would be forcibly moved to the original submenu menu trigger item.  If we only get rid of `onExit`, then `Case 1` would lose focus to the body instead.

Like I mention in the comment above, the ideal logic would be that focus should always move to the menu item that the user hovers and that FocusScope wouldn't then shift focus anywhere as menus are closed/opened as a result of the hover interaction the user performed. This `onExit` approach was the best I could come up with since it happens after menus close and after FocusScope does its focus shifting.

Reply:
resolved after offline discussion. Made menuItems always have "tabIndex=-1" and got rid of FocusScope containment at every menu level